src/lib.rs

```diff
@@ -678,15 +680,21 @@ pub fn over_in(src: u32, dst: u32, alpha: u32) -> u32 {
     return (((src_rb + dst_rb) >> 8) & mask) | ((src_ag + dst_ag) & !mask);
 }
 
-#[cfg(not(any(target_arch = "x86", target_arch = "x86_64")))]
+#[cfg(not(all(any(target_arch = "x86", target_arch = "x86_64"), target_feature = "sse2")))]
 pub fn over_in_row(src: &[u32], dst: &mut [u32], alpha: u32) {
     for (dst, src) in dst.iter_mut().zip(src) {
         *dst = over_in(*src, *dst, alpha as u32);
     }
 }
 
-#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+#[cfg(all(any(target_arch = "x86", target_arch = "x86_64"), target_feature = "sse2"))]
 pub fn over_in_row(src: &[u32], dst: &mut [u32], alpha: u32) {
+    #[cfg(target_arch = "x86")]
+    use std::arch::x86::{
+        _mm_loadu_si128,
+        _mm_storeu_si128,
+    };
+    #[cfg(target_arch = "x86_64")]
```

Comment:
This piece should use the same x86_intrinsics stuff from below. Other than that this looks good.

Reply:
Good point.